MITADS/opensubtitles_exporter.py

```diff
@@ -0,0 +1,367 @@
+from unidecode import unidecode  
+import time
+
+global start_time_regexp
+
+mapping_normalization = [
+
+  ## If the sentence start with a number, the sentence is removed
+  [ re.compile('^\d+(.*)'), u'' ],  
+
+  ## Remove the stuff inside ()[]{}
+  [ re.compile('(\(|\[|{)[^(\)|\]|})]*(\)|\]|})'), u'' ],
+  ##must be twice time for nested parentheses
+  [ re.compile('(\(|\[|{)[^(\)|\]|})]*(\)|\]|})'), u'' ],
+ 
+  ##remove uninteresting characters
+  [ re.compile('\-|=|_|–|\+|\(|\||—|\)|\[|\]|~|\*|/|"|¨|\^'), u' ' ],     
+  
+  [ u'¢' , u'c' ],  
+  
+  ## Sanitize ... to . 
+  # Attention: Ellipsis are part of grammar, they are punctuation, they should not be eliminated
+  [ re.compile('\.+'), u'.' ],
+  ##normalize spaces
+  [ re.compile('\n|\t|\r|\s+'), u' ' ],
+
+  ##accentate maiuscole
+  [ re.compile('È'), u'e\'' ],
+
+  ##Sanitize single apex
+  [ re.compile('´|`|\'\''), u'\'' ],
+
+  ## To avoid conflicts with single ' and accented letter we removed them
+  [ re.compile('(\s|^)(\')([^\']*)(\')(\s|$)'), r'\3' ],
+
+  ## remove char for those cases
+  [ re.compile('(#\d+)|#|(\s°)'), u'' ],
+  
+  ## Sanitization for those cases
+  [ u'n°' , u'numero ' ],  
+
+  ## Sanitization for currency values
+  [ re.compile('\$\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\1 dollari' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*\$'), r'\1 dollari' ],
+  [ re.compile('(₤|£)\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\2 lire' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*₤'), r'\1 lire' ],
+  [ re.compile('(€)\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\2 euro' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*€'), r'\1 euro' ],
+
+  ##space trim - it must be reapplied again
+  [ re.compile('\s+'), u' ' ]
+]
+
+
+mapping_normalization_after_decode = [
+
+  ## Convert old fashion accented letter to the real accented letter
+  [ u'E\' ', u'è ' ],
+  [ re.compile('a\'(\s|$|,|\.|\?)'), r'à\1' ], 
+  [ re.compile('e\'(\s|$|,|\.|\?)'), r'è\1' ],
+  [ re.compile('i\'(\s|$|,|\.|\?)'), r'ì\1' ],
+  [ re.compile('o\'(\s|$|,|\.|\?)'), r'ò\1' ],
+  [ re.compile('u\'(\s|$|,|\.|\?)'), r'ù\1' ],     
+
+]
+
+
+def maybe_normalize(value, mapping=mapping_normalization):
+  for norm in mapping:
+    if(value==''):
+      break
+
+    if type(norm[0]) == str:
+      value = value.replace(norm[0], norm[1])
+    elif isinstance(norm[0], Pattern):
+      try:
+        value = norm[0].sub(norm[1], value)
+      except Exception as e:
+        raise(e)
+    else:
+      print('UNEXPECTED', type(norm[0]), norm[0])
+
+  return value
+
+def line_not_relevant(text):
+  
+  if(len(text)<2):
+    return True
+
+  
+  ## If the sentence include those symbols we ignore it (is part of the opening subtitles)  
+  if('®' in text
+     or '©' in text
+     or '±' in text):
+    return True
+
+  ##sentence with one word that contain those symbols, we ignore it
+  if(not ' ' in text and 
+    (   '_' in text
+     or '@' in text
+     or '}' in text     
+     or '+' in text
+     or '{' in text)):
+     return True
```

Comment:
I think is better to not have single words because have a lot of false positives like names so is better to replace with `if(not ' ' in text):`

Reply:
conversations are full of single words. There are also many single words with '!' Or '?' , higher than those of the subtitles. ex: "Ciao!", "AHAHAH !!", "Allora?".  
A middle way is to delete single words with '!?'  together ?